src/connector/mysql/conversion.rs

```diff
@@ -68,31 +68,52 @@ impl TakeRow for my::Row {
                     let date = NaiveDate::from_ymd(year.into(), month.into(), day.into());
                     let dt = NaiveDateTime::new(date, time);
 
-                    Value::DateTime(DateTime::<Utc>::from_utc(dt, Utc))
+                    Value::datetime(DateTime::<Utc>::from_utc(dt, Utc))
                 }
                 #[cfg(feature = "chrono-0_4")]
                 my::Value::Time(is_neg, days, hours, minutes, seconds, micros) => {
                     if is_neg {
-                        return Err(crate::error::Error::builder(ErrorKind::ConversionError(
-                            "Failed to convert a negative time",
-                        ))
-                        .build());
+                        let kind = ErrorKind::ConversionError("Failed to convert a negative time");
+                        return Err(crate::error::Error::builder(kind).build());
                     }
 
                     if days != 0 {
-                        return Err(crate::error::Error::builder(ErrorKind::ConversionError(
-                            "Failed to read a MySQL `time` as duration",
-                        ))
-                        .build());
+                        let kind = ErrorKind::ConversionError("Failed to read a MySQL `time` as duration");
+                        return Err(crate::error::Error::builder(kind).build());
                     }
 
                     let time = NaiveTime::from_hms_micro(hours.into(), minutes.into(), seconds.into(), micros);
-
-                    Value::DateTime(DateTime::<Utc>::from_utc(
-                        NaiveDateTime::new(NaiveDate::from_ymd(1970, 1, 1), time),
-                        Utc,
-                    ))
+                    Value::time(time)
                 }
+                my::Value::NULL => match column.column_type() {
+                    MYSQL_TYPE_DECIMAL | MYSQL_TYPE_FLOAT | MYSQL_TYPE_DOUBLE | MYSQL_TYPE_NEWDECIMAL => {
+                        Value::Real(None)
+                    }
+                    MYSQL_TYPE_NULL => Value::Integer(None),
+                    MYSQL_TYPE_TINY | MYSQL_TYPE_SHORT | MYSQL_TYPE_LONG | MYSQL_TYPE_LONGLONG => Value::Integer(None),
+                    #[cfg(feature = "chrono-0_4")]
+                    MYSQL_TYPE_TIMESTAMP
+                    | MYSQL_TYPE_TIME
+                    | MYSQL_TYPE_DATE
+                    | MYSQL_TYPE_DATETIME
+                    | MYSQL_TYPE_YEAR
+                    | MYSQL_TYPE_NEWDATE
+                    | MYSQL_TYPE_TIMESTAMP2
+                    | MYSQL_TYPE_DATETIME2
+                    | MYSQL_TYPE_TIME2 => Value::DateTime(None),
+                    MYSQL_TYPE_VARCHAR | MYSQL_TYPE_VAR_STRING | MYSQL_TYPE_STRING => Value::Text(None),
```

Comment:
I think we have to check the encoding here to determine if it's text or bytes. Same for the blob variants.

Reply:
Can you be a bit more elaborate? What more should we check here?

Reply:
It's the `column.character_set()` thing I hit in the type mapping work. This line on master: https://github.com/prisma/quaint/blob/c2b2b4af698e2440d81e4fd2820a787ab42ca56a/src/connector/mysql/conversion.rs#L51

Reply:
Would this be enough? https://github.com/prisma/quaint/pull/131/commits/35d635dfaa565571f8e9afa35e83748c071892bd